Make edits to the demo data section (#3020)
influxdata/docs-v2: content/influxdb/cloud/write-data/developer-tools/line-protocol.md

---
title: Write data with line protocol
description: >
  Use line protocol to write data to InfluxDB. 
menu:
  influxdb_cloud:
    name: Write with line protocol
    parent: Developer tools
weight: 204
related:
  - /influxdb/cloud/reference/syntax/line-protocol/
  - /influxdb/cloud/reference/cli/influx/write/
---

> Use line protocol to write data to InfluxDB Cloud. Line protocol is a is a text-based format that provides the measurement, tag set, field set, and timestamp of a data point. Learn more about [line protocol](/influxdb/v2.0/reference/syntax/line-protocol/). 

Reviewer: ```suggestion
Use line protocol to write data to InfluxDB Cloud. Line protocol is a text-based format that provides the measurement, tag set, field set, and timestamp of a data point. Learn more about [line protocol](/influxdb/v2.0/reference/syntax/line-protocol/). 
```

Write data using line protocol with the following methods: 
- [UI](#write-line-protocol-with-ui)
- [Influx write command](#influx-write-command)

## Write line protocol with UI 

To write data using Cloud's line protocol file upload, do the following: 

1. Verify the line protocol adheres to the following conventions:  
   - Each line represents a data point.
   - Each data point requires a:  
     - [*measurement*](/influxdb/cloud/reference/syntax/line-protocol/#measurement)
     - [*field set*](/influxdb/cloud/reference/syntax/line-protocol/#field-set)
     - (Optional) [*tag set*](/influxdb/cloud/reference/syntax/line-protocol/#tag-set) 
     - [*timestamp*](/influxdb/cloud/reference/syntax/line-protocol/#timestamp) 
2. Click **Data** in your left navigation bar. 
3. Select **Line Protocol** and choose your bucket. 

> 4. Select your **Precision** in the dropdown menu. By default, the precision is set to nanosections. 

Reviewer: ```suggestion
4. Select your **Precision** in the dropdown menu. By default, the precision is set to nanoseconds. 
```

5. Do one of the following: 
   - Upload your line protocol file. 
   - Write your line protocol manually, and then click **Submit**. 

> ## Influx write command

Reviewer: ```suggestion
## Write line protocol using the influx write command
```

Use the [`influx write` command](/influxdb/v2.0/reference/cli/influx/write/) to write data using line protocol
to InfluxDB. 

> Set the path for your `--file` flag and set the path for your `--format`. 

Reviewer: ```suggestion
Set the path to the line protocol file using the `-f` flag.
```

> Verify that your line protocol follows the correct conventions. 

Reviewer: ```suggestion
Verify that your line protocol follows the correct [line protocol format](#line-protocol-format) conventions. 
```

#### Example write command
```sh

> influx write -b example-bucket -f path/to/example.csv

Reviewer: ```suggestion
influx write -b example-bucket -f path/to/example.txt 
```
